capstone/src/main/java/com/google/sps/servlets/DeleteFlags.java

```diff
@@ -38,37 +38,42 @@
 @WebServlet("/delete-flag")
 public class DeleteFlags extends HttpServlet {
 
-  private static final DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
+    private static final DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
 
-  @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-    Query query = new Query("Flag").addSort("date", SortDirection.ASCENDING);
+        Query query = new Query("Flag").addSort("date", SortDirection.ASCENDING);
 
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-	PreparedQuery results = datastore.prepare(query);
-    Date currentDate = new Date();
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+        Date currentDate = new Date();
 
-    /** Use a calendar object to count the date when the flags
-      * should expire. 
-      */
-    Calendar c = Calendar.getInstance();
-    c.setTime(currentDate);
-    c.add(Calendar.DATE, -14);
-    Date currentDateMinusTwoWeeks = c.getTime();
+        /** delete flags if the date property of the flag is before
+         * the previously calculated date. 
+         */
+        for (Entity entity: results.asIterable()) {
+            Date flagDate = (Date) entity.getProperty("date");
+            if (isTooOld(flagDate, currentDate)) {
+                datastore.delete(entity.getKey());
+            } else {
+                break;
+            }
+        }
+    }
 
-    /** delete flags if the date property of the flag is before
-      * the previously calculated date. 
-      */
-    for (Entity entity: results.asIterable()) {
-        Date flagDate = (Date)entity.getProperty("date");
+    public boolean isTooOld(Date flagDate, Date currentDate) {
+        Calendar c = Calendar.getInstance();
+        c.setTime(currentDate);
+        c.add(Calendar.DATE, -14);
+        Date currentDateMinusTwoWeeks = c.getTime();
         if (currentDateMinusTwoWeeks.compareTo(flagDate) > 0) {
-            datastore.delete(entity.getKey());
-        } else {
-            break;
+            return true;
         }
+        return false;
```

Comment:
Why not just return false at the end of the if?

Reply:
Discussed offline. Can just return currentDateMinusTwoWeeks.compareTo(flagDate) > 0